indexer-app/src/cmr/indexer/services/index_service.clj

```diff
@@ -449,13 +458,27 @@
           ;; delete concept from primary concept index
           (do
             (es/delete-document
-             context index-names (concept-mapping-types concept-type)
-             concept-id revision-id elastic-version elastic-options)
-            ;; propagate collection deletion to granules and variables
+              context index-names (concept-mapping-types concept-type)
+              concept-id revision-id elastic-version elastic-options)
+            ;; Index a deleted-granule document when granule is deleted
+            (when (= :granule concept-type)
+              (let [es-doc (dg/deleted-granule->elastic-doc concept)]
```

Comment:
NIT: refactor this `let` block into a function in `cmr.indexer.data.concepts.deleted-granule` namespace, call it something like `insert-deleted-granule`.